wgpu/src/lib.rs

```diff
@@ -3887,7 +3901,17 @@ impl CommandEncoder {
     /// Begins recording of a compute pass.
     ///
     /// This function returns a [`ComputePass`] object which records a single compute pass.
-    pub fn begin_compute_pass(&mut self, desc: &ComputePassDescriptor<'_>) -> ComputePass {
+    ///
+    /// As long as the returned  [`ComputePass`] has not ended,
+    /// any mutable operation on this command encoder causes an error and invalidates it.
+    /// Note that the lifetime constraint protects against this, but it is possible to opt out of it
+    /// by calling [`ComputePass::make_static`].
+    /// This can be useful for runtime handling of the encoder->pass
```

Comment:
However this is phrased exactly, I think `begin_render_pass` and `begin_compute_pass` should have **exactly the same wording**. The benefits of this are that:

* Users don't need to wonder if there's something subtly semantically different implied by the two wordings.
* Future edits are more likely to be done consistently in a find-and-replace fashion.

Reply:
I agree in principle, but this was a deliberate choice for the moment: right now renderpass & computepass _are_ fundamentally different since renderpass doesn't take (shared) ownership of its encoder & passed in resources during recording like computepass does now.
Ideally, before 0.21 gets ready both behave the same and then we can c&p this over. I'll leave a todo note!

Reply:
mhh probably easier to just have it be the same and then leave a note that the lifetime removal method is not yet available, less headache